### PR TITLE
Added interface for per request error reporting

### PR DESCRIPTION
**Pull request by vdantu**
Before or while filing an issue please feel free to join our [ slack channel](https://join.slack.com/t/mms-awslabs/shared_invite/enQtNDk4MTgzNDc5NzE4LTBkYTAwMjBjMTVmZTdkODRmYTZkNjdjZGYxZDI0ODhiZDdlM2Y0ZGJiZTczMGY3Njc4MmM3OTQ0OWI2ZDMyNGQ) to get in touch with development team, ask questions, find out what's cooking and more!

## Issue #, if available:
fixes: https://github.com/awslabs/mxnet-model-server/issues/757

## Testing done:

**To run CI tests on your changes refer [README.md](https://github.com/awslabs/mxnet-model-server/blob/master/ci/README.md)**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/server/src/main/java/com/amazonaws/ml/mms/util/messages/Predictions.java [hunk 1]
```diff
@@ -15,6 +15,9 @@
 public class Predictions {
 
     private String requestId;
+    private int code;
```
Comment (frankfliu) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/util/messages/Predictions.java]: better name it statusCode

File: frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/Job.java [hunk 2]
```diff
@@ -72,9 +73,17 @@ public void setScheduled() {
         scheduled = System.currentTimeMillis();
     }
 
-    public void response(byte[] body, CharSequence contentType) {
+    public void response(
+            byte[] body, CharSequence contentType, int statusCode, String statusPhrase) {
         FullHttpResponse resp =
-                new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK, false);
+                new DefaultFullHttpResponse(
```
Comment (frankfliu) [inline frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/Job.java]: HttpResponseStatus.valueOf() will create a HttPResponseStatus object, which is not necessary:

            HttpResponseStatus status;
            if (reasonPhrase == null) {
                status = HttpResponseStatus.valueOf(statusCode);
            } else {
                status = HttpResponseStatus.valueOf(statusCode, reasonPhrase);
            }

Comment (vdantu) [reply frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/Job.java]: Done.

File: frontend/server/src/main/java/com/amazonaws/ml/mms/util/messages/Predictions.java (hunk 1 above)
Comment (vdantu) [reply frontend/server/src/main/java/com/amazonaws/ml/mms/util/messages/Predictions.java]: done

**Pull request merged**